script/tool/lib/src/common/plugin_command.dart

```diff
@@ -374,14 +418,10 @@ abstract class PluginCommand extends Command<void> {
     return gitVersionFinder;
   }
 
-  // Returns packages that have been changed relative to the git base.
-  Future<Set<String>> _getChangedPackages() async {
-    final GitVersionFinder gitVersionFinder = await retrieveVersionFinder();
-
-    final List<String> allChangedFiles =
-        await gitVersionFinder.getChangedFiles();
+  // Returns packages that have been changed given a list of changed files.
+  Set<String> _getChangedPackages(List<String> changedFiles) {
     final Set<String> packages = <String>{};
-    for (final String path in allChangedFiles) {
+    for (final String path in changedFiles) {
       final List<String> pathComponents = path.split('/');
```

Comment:
In practice it's not because this is only ever called with `git` output, which always uses POSIX paths. I changed this to `p.posix.split` and added a comment to the method saying that they must be POSIX paths as a quick improvement. (The better, more involved solution would be to use `File`s rather than `String`s.)